app/frontend/src/features/auth/authActions.ts

```diff
@@ -43,3 +43,12 @@ export const signup = createAsyncThunk(
     return { token: auth.token, jailed: auth.jailed, user: null };
   }
 );
+
+export const tokenLogout = createAsyncThunk(
+  "auth/logout",
+  async (sessionToken: string) => {
+    await service.user.logout(sessionToken);
+
+    return { token: null, jailed: null, user: null };
```

Comment:
The return value is passed to the reducer, you won't actually need to return anything (you will just set null if logging out).